React/Modules/RCTUIManager.m

```diff
@@ -723,6 +727,20 @@ - (void)_amendPendingUIBlocksWithStylePropagationUpdateForShadowView:(RCTShadowV
   }
 }
 
+- (void)addUIManagerObserver:(id<RCTUIManagerObserver>)observer
+{
+  [_uiManagerObserversLock lock];
```

Comment:
I *think* it would also be safe to remove the lock and replace it with dispatch_async(main queue) since the main queue is a serial queue.